packages/engine/src/interaction/systems/EquippableSystem.ts

```diff
@@ -8,29 +8,33 @@ import { AvatarComponent } from '../../avatar/components/AvatarComponent'
 import { LocalInputTagComponent } from '../../input/components/LocalInputTagComponent'
 import { dispatchFrom } from '../../networking/functions/dispatchFrom'
 import { System } from '../../ecs/classes/System'
-import { Not } from 'bitecs'
 import { World } from '../../ecs/classes/World'
 import { NetworkWorldAction } from '../../networking/functions/NetworkWorldAction'
 import matches from 'ts-matches'
 import { useWorld } from '../../ecs/functions/SystemHooks'
 import { teleportRigidbody } from '../../physics/functions/teleportRigidbody'
-
-const networkUserQuery = defineQuery([Not(LocalInputTagComponent), AvatarComponent, TransformComponent])
-const equippableQuery = defineQuery([EquipperComponent])
+import { Engine } from '../../ecs/classes/Engine'
+import { equipEntity, getParity } from '../functions/equippableFunctions'
+import { isClient } from '../../common/functions/isClient'
+import { EquippedComponent } from '../components/EquippedComponent'
 
 function equippableActionReceptor(action) {
-  matches(action).when(NetworkWorldAction.setEquippedObject.matches, (a) => {
-    // const world = useWorld()
-    // // const equipper = world.getUserAvatarEntity(a.$userId)
-    // // const equipped = world.getNetworkObject(a.$userId, a.networkId)
-    // if (!equipped) {
-    //   return console.warn(
-    //     `Equipped entity with id ${equipped} does not exist! You should probably reconnect...`
-    //   )
-    // }
-    // if (a.equip) {
-    //   equipEntity(equipper, equipped)
-    // } else {
+  const world = useWorld()
+
+  matches(action).when(NetworkWorldAction.setEquippedObject.matchesFromAny, (a) => {
```

Comment:
what's the difference between `.matches` and `.matchesFromAny`? i always forget ><

Reply:
`.matches` will match an action only when it comes from a trusted source (e.g., the host, or itself)
`.matchesFromAny` will match an action no matter who sent it

Reply:
@NateTheGreatt happy to rename these for clarity if you can think of better names